admiral/pkg/clusters/handler.go

```diff
@@ -557,14 +560,11 @@ func addUpdateVirtualService(obj *v1alpha3.VirtualService, exist *v1alpha3.Virtu
 	if exist == nil || len(exist.Spec.Hosts) == 0 {
 		obj.Namespace = namespace
 		obj.ResourceVersion = ""
-		_, err = rc.VirtualServiceController.IstioClient.NetworkingV1alpha3().VirtualServices(namespace).Create(obj)
+		_, err = rc.VirtualServiceController.IstioClient.NetworkingV1alpha3().VirtualServices(namespace).Create(ctx, obj, v12.CreateOptions{})
 		op = "Add"
 	} else {
-		exist.Labels = obj.Labels
-		exist.Annotations = obj.Annotations
-		exist.Spec = obj.Spec
```

Comment:
hey @aattuluri , we were getting `copylock` linting errors on this line and @nirvanagit and I quickly scanned through the code and it seemed we could pass `obj` directly to the `Update` func without having to assign it to `exist`. Since we don't have much historic context to it, can you please review this and let us know if this is ok?

Reply:
In this case obj is not something returned by the client but rather a new generated one. I think it might work. I am fine as long as the tests pass. Dont really remember why it was written this way :)